### ISSUE TITLE
AFK Fishing issue

### ISSUE THREAD

**Issue by Parshath20** [Hackswell/GloryOfEfficiency#2]
When using AFK Fishing, once you reel in any item(fish, algae, trash) an empty treasure inventory window opens and the item caught disappears. The window can be closed and the AFK Fishing resumes. This occurs with the AFK Fishing options either on or off. 

When using Auto Loot Treasures options, the empty treasure inventory window flashes open then closes but the caught item is still lost. 

Using SMAPI 4.0.2

EDIT: Grabbed from SMAPI when activating AFK Fishing:

```
[JoysOfEfficiency] This mod failed in the GameLoop.UpdateTicked event. Technical details:
InvalidCastException: Unable to cast object of type 'System.Collections.Generic.Dictionary`2[System.String,StardewValley.GameData.Locations.LocationData]' to type 'System.Collections.Generic.Dictionary`2[System.String,System.String]'.
   at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T](T existingInstance) in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentReader.cs:line 176
   at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T]() in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentReader.cs:line 152
   at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T]() in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentReader.cs:line 61
   at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](String assetName, Action`1 recordDisposableObject) in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 324
   at Microsoft.Xna.Framework.Content.ContentManager.Load[T](String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 255
   at StardewModdingAPI.Framework.ContentManagers.BaseContentManager.RawLoad[T](IAssetName assetName, Boolean useCache) in SMAPI\Framework\ContentManagers\BaseContentManager.cs:line 333
   at StardewModdingAPI.Framework.ContentManagers.GameContentManager.LoadExact[T](IAssetName assetName, Boolean useCache)
   at StardewValley.LocalizedContentManager.Load[T](String assetName, LanguageCode language) in D:\GitlabRunner\builds\Gq5qA5P4\0\ConcernedApe\stardewvalley\Farmer\Farmer\LocalizedContentManager.cs:line 524
   at JoysOfEfficiency.Huds.FishingProbabilitiesBox.GetFishes(Int32 waterDepth, Farmer who, String locationName) in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/Huds/FishingProbabilitiesBox.cs:line 101
   at JoysOfEfficiency.Huds.FishingProbabilitiesBox.GetFishesFarm(Int32 waterDepth, Farmer who) in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/Huds/FishingProbabilitiesBox.cs:line 257
   at JoysOfEfficiency.Huds.FishingProbabilitiesBox.GetFishes(GameLocation location, Int32 bait, Int32 waterDepth, Farmer who, Int32 trial) in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/Huds/FishingProbabilitiesBox.cs:line 70
   at JoysOfEfficiency.Huds.FishingProbabilitiesBox.UpdateProbabilities(FishingRod rod) in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/Huds/FishingProbabilitiesBox.cs:line 43
   at JoysOfEfficiency.EventHandler.UpdateEvents.OnEveryUpdate() in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/EventHandler/UpdateEvents.cs:line 53
   at JoysOfEfficiency.EventHandler.UpdateEvents.OnGameUpdateEvent(Object sender, UpdateTickedEventArgs args) in /home/woody/working/JoysOfEfficiency/JoysOfEfficiency/EventHandler/UpdateEvents.cs:line 29
   at StardewModdingAPI.Framework.Events.ManagedEvent`1.Raise(TEventArgs args) in SMAPI\Framework\Events\ManagedEvent.cs:line 101
```

**Comment by Hackswell** [comment]
Please try ALPHA3 pre-release.

**Comment by Exaxxion** [comment]
Confirmed NOT fixed in 1.5.0alpha3. Auto-Fishing still deletes non-fish items pulled from water.
